### DISCUSSION TITLE
[Feature]: Save upload concurrency instead of resetting to 2

### DISCUSSION THREAD

**Discussion by ThreeAurora** [immich-app/immich#9173] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [x] Yes

### The feature

When the web page is refreshed, the value of Upload Concurrency is reset to 2.

So every time I upload some files, I have to modify its value.

Is it possible to save upload concurrency?

Thanks.

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by Matesaktesak** [comment]
This would be very much appreciated!

**Reply by jadster** [reply]
Yes please!

**Comment by pssolanki111** [comment]
Bumping this up. Increasing the concurrent uploads counts greatly increases the overall upload speed.

**Reply by bo0tzz** [reply]
"Bumping" doesn't accomplish anything other than noise. Please just upvote the original post if you don't have more to add.

**Comment by l4rm4nd** [comment]
I recommend adding a configuration parameter or environment variable to hardcode the default concurrency.

https://github.com/immich-app/immich/discussions/15307

**Reply by dodancs** [reply]
Or just put it into user settings, so that each user can set their own value (someone might not want to fully saturate their connection). And maybe add a "max-upload-concurrency" setting as well to set a limit for the whole server.

**Reply by ssjrocks** [reply]
> I recommend adding a configuration parameter or environment variable to hardcode the default concurrency.
> 
> #15307

second... ive done this already for my own server, save 3 clicks per large folder which for me and my 540 album import, that needs to be done folder by folder to curate faces and albums meticulously, this has been a massive improvement to my experience

**Reply by ssjrocks** [reply]
> Or just put it into user settings, so that each user can set their own value (someone might not want to fully saturate their connection). And maybe add a "max-upload-concurrency" setting as well to set a limit for the whole server.

for multi users setups this would just make sense
